### PR TITLE
Restrict logging of errors with verbosity level

### PR DESCRIPTION
**Pull request by andrewdacenko**
Allow control errors logging with PDFJS.verbosity level

### CODE REVIEW

**Comment by timvandermeij** [comment]
Can't we simplify this a lot by keeping a similar structure as https://github.com/mozilla/pdf.js/blob/fa0f09bcf02581689e80e612c28536787f37c7d3/src/shared/util.js#L196 ? I see no need for another fuction and the bindings...

**Comment by andrewdacenko** [comment]
@timvandermeij 

In my opinion this code looks much more elegant

```javascript
// Fatal errors.
function err(fn) {
  if (PDFJS.verbosity >= PDFJS.VERBOSITY_LEVELS.errors) {
    fn();
  }
}

// Fatal errors that should trigger the fallback UI and halt execution by
// throwing an exception.
function error(msg) {
  // If multiple arguments were passed, pass them all to the log function.
  if (arguments.length > 1) {
    var logArguments = ['Error:'];
    logArguments.push.apply(logArguments, arguments);
    err(console.log.apply.bind(console.log, console, logArguments));
    // Join the arguments into a single string for the lines below.
    msg = [].join.call(arguments, ' ');
  } else {
    err(console.log.bind(console, 'Error: ' + msg));
  }
  err(console.log.bind(console, backtrace()));
  UnsupportedManager.notify(UNSUPPORTED_FEATURES.unknown);
  throw new Error(msg);
}
```

than this one

```javascript
// Fatal errors that should trigger the fallback UI and halt execution by
// throwing an exception.
function error(msg) {
  // If multiple arguments were passed, pass them all to the log function.
  if (arguments.length > 1) {
    var logArguments = ['Error:'];
    logArguments.push.apply(logArguments, arguments);
    if (PDFJS.verbosity >= PDFJS.VERBOSITY_LEVELS.errors) {
      console.log.apply(console, logArguments);
    }
    // Join the arguments into a single string for the lines below.
    msg = [].join.call(arguments, ' ');
  } else {
    if (PDFJS.verbosity >= PDFJS.VERBOSITY_LEVELS.errors) {
      console.log('Error: ' + msg);
    }
  }
  if (PDFJS.verbosity >= PDFJS.VERBOSITY_LEVELS.errors) {
    console.log(backtrace());
  }
  UnsupportedManager.notify(UNSUPPORTED_FEATURES.unknown);
  throw new Error(msg);
}
```

Please, let me now, which you prefer.

**Comment by yurydelendik** [comment]
> Please, let me now, which you prefer.

Neither solution above look good. Can you review calls of error() function and make it as a single argument function (similar to info/warn)? If multiple arguments are used, make it something like `error(formatString("test", 1))` and add an utility function formatString.

Currently, the patch contains `err()` which is not really useful by itself.

**Comment by Snuffleupagus** [comment]
> Can you review calls of error() function and make it as a single argument function (similar to info/warn)?

A very quick look through the code-base seem to indicate that no caller of `error` passes in more than *one* argument, but a thorough review is **obviously** necessary.
A fair number of callers actually pre-concatenate strings themselves, before calling `error`, so I fully support the idea of changing `error` to only accept a *single* argument.

**Comment by andrewdacenko** [comment]
@yurydelendik , @timvandermeij , @Snuffleupagus Reviewed code and found no caller with multiple arguments. Changed code to simple
```javascript
// Fatal errors that should trigger the fallback UI and halt execution by
// throwing an exception.
function error(msg) {
  if (PDFJS.verbosity >= PDFJS.VERBOSITY_LEVELS.errors) {
    console.log(msg);
    console.log(backtrace());
  }
  UnsupportedManager.notify(UNSUPPORTED_FEATURES.unknown);
  throw new Error(msg);
}
```
multiple console logs for separate lines. Or can be done with "\n".

File: src/shared/util.js [hunk 1]
```diff
@@ -208,17 +208,10 @@ function warn(msg) {
...
-  if (arguments.length > 1) {
-    var logArguments = ['Error:'];
-    logArguments.push.apply(logArguments, arguments);
-    console.log.apply(console, logArguments);
-    // Join the arguments into a single string for the lines below.
-    msg = [].join.call(arguments, ' ');
-  } else {
-    console.log('Error: ' + msg);
+  if (PDFJS.verbosity >= PDFJS.VERBOSITY_LEVELS.errors) {
+    console.log(msg);
```
Comment (Snuffleupagus) [inline src/shared/util.js]: This line should be `console.log('Error: ' + msg);`.

**Comment by Snuffleupagus** [comment]
@andrewdacenko Please squash the commits into one, see: https://github.com/mozilla/pdf.js/wiki/Squashing-Commits.

**Comment by andrewdacenko** [comment]
@Snuffleupagus Done. 

@timvandermeij , @yurydelendik thanks for your attention )

**Comment by Snuffleupagus** [comment]
Thanks for the patch!

**Pull request merged**